### ISSUE TITLE
aria-role should be reserved for 'purpose' not 'form-factor'

### ISSUE THREAD

**Issue by Meteor0id** [w3c/aria#804]
I just learned about using ARIA in html. The first thing I decided to try was properly mark up the navigation of a website. The main navigation is done using one horizontal navigation bar. However, I can not mark it up with both the role 'navigation', and the role 'menu'. The navigation exists only out of one division with anchors as its immediate children.

I seems odd to me that there should be a distinction between a menu bar and a navigation element. The element has the purpose navigation, but in the form of a menu bar. It seems to me ARIA is not properly differentiating between ´intent´ and ´form of presentation´.

A "menubar" in my mind is not a valid 'role'. A menubar is a form factor, which can be atributed to some element. That element might happen to have the 'role' of providing "navigation".

Of course I could simply mark this navigation bar up with the role "navigation" and leave "menubar" out of it. However, how will I mark up the children of this navigation element? And how do I amrk up its orientation is horizontal? (For menubar I can use aria-orientation, but for navigation I can not specify this form detail).

**Comment by mcking65** [comment]
@Meteor0id commented:

>It seems odd to me that there should be a distinction between a menu bar and a navigation element. 

Broadly speaking, ARIA has two kinds of roles: structures and widgets. Structural roles describe the semantics of elements that are not interactive, e.g., a list or heading. Widget roles describe the semantics and interaction model of interactive elements, e.g., tabs and menus.

The navigation role is a special kind of structure called a landmark region. So, to understand its purpose, you will need to first learn what landmark regions do for users and how to implement them in order to achieve their intended benefits to users. To start, read the [landmark regions section of ARIA authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_landmark).

A menubar is a widget. The use of a widget role carries some specific requirements to make the widget conform to expectations for various behaviorsk, including correct communications of states, properties, and the provision of keyboard interaction models. To get started, please read the [Read Me First](https://www.w3.org/TR/wai-aria-practices-1.1/#read_me_first) section of the authoring practices guide.

>The element has the purpose navigation, but in the form of a menu bar.

You might call it a menubar, but that does not mean that it is an ARIA menubar widget. It may be a few links at the top of a page that are use for navigation, which would simply be a list of navigation links.

>It seems to me ARIA is not properly differentiating between ´intent´ and ´form of presentation´.

ARIA describes the meaning of form of presentation, not the form of presentation.

>A "menubar" in my mind is not a valid 'role'. A menubar is a form factor, which can be atributed to some element. That element might happen to have the 'role' of providing "navigation".

ARIA is a bout translating the meaning of presentation into the language of accessibility APIs that live in operating systems. Then, assistive technologies, such as screen readers, suck the information out of the accessibility APIs and make it understandable by people who use screen readers. These accessibility APIs, that are different on Windows, Mac, Linux, iOS, Android, ChromeOS, ... all have very specific ways of describing elements. So, when we use words like "menu" and "role" in ARIA, they are couched in the need to communicate via APIs. We have tried to translate this into the language of web engineers in the ARIA Authoring Practices.

>Of course I could simply mark this navigation bar up with the role "navigation" and leave "menubar" out of it.

That is very likely best if it is just a collection of links.

>However, how will I mark up the children of this navigation element?

If they are links, please just leave them as HTML links.

>And how do I amrk up its orientation is horizontal? (For menubar I can use aria-orientation, but for navigation I can not specify this form detail).

It is only necessary to use aria-orientation if you are using a widget role where arrow key navigation is expected. Then, aria-orientation helps a screen reader know if users should use left/right or up/down to navigate the choices.

**Comment by carmacleod** [comment]
If you truly want to use menubar because you are implementing the keyboard interaction described for the menubar role, then add a `<nav>` parent. Then you will have a "navigation region containing a menubar". (Of course, if you have more than one nav on the page, give each one a label using either `aria-label` or `aria-labelledby`).

**Comment by jnurthen** [comment]
@Meteor0id - As @mcking65 has conveyed by pointing to information in the ARIA Authoring Practices Guide I think this issue would be more appropriate for the APG. I don't think there is a request here for the spec to be modified. If you disagree please reopen this issue. 
If the information in the authoring practices does not help clarify and you believe there is something else that is needed there then please log an issue against the ARIA Authoring Practices at https://github.com/w3c/aria-practices
